iceoryx_posh/source/popo/ports/server_port_user.cpp

```diff
@@ -61,20 +71,72 @@ bool ServerPortUser::hasLostRequestsSinceLastCall() noexcept
 }
 
 cxx::expected<ResponseHeader*, AllocationError>
-ServerPortUser::allocateResponse(const uint32_t /*userPayloadSize*/) noexcept
+ServerPortUser::allocateResponse(const RequestHeader* const requestHeader,
+                                 const uint32_t userPayloadSize,
+                                 const uint32_t userPayloadAlignment) noexcept
 {
-    /// @todo
-    return cxx::error<AllocationError>(AllocationError::RUNNING_OUT_OF_CHUNKS);
+    cxx::Ensures(requestHeader != nullptr && "requestHeader must not be a nullptr");
```

Comment:
What about returning `cxx::error` with an allocation error like invalid argument? Or event better we get the request header as reference and we can remove those lines and can guarantee that the request header is always not null?

Reply:
Good point. I didn't thought about extending `AllocationError` and was hesitant to introduce another error enum but extending the `AllocationError` enum would be the best option